src/redisearch_rs/sorting_vector/src/tests.rs

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use std::ffi::{CString, c_char};
+
+use icu_casemap::CaseMapper;
+
+use super::{RSSortingVector, RSValueTrait};
+
+#[derive(Clone, Debug, PartialEq)]
+enum RSValueMock {
+    Null,
+    Number(f64),
+    String(String),
+    Reference(Box<RSValueMock>),
+}
+
+impl RSValueTrait for RSValueMock {
+    fn create_null() -> Self {
+        RSValueMock::Null
+    }
+
+    fn create_string(s: &str) -> Self {
+        RSValueMock::String(s.to_string())
+    }
+
+    fn create_num(num: f64) -> Self {
+        RSValueMock::Number(num)
+    }
+
+    fn create_ref(value: Self) -> Self {
+        RSValueMock::Reference(Box::new(value))
+    }
+
+    fn is_null(&self) -> bool {
+        matches!(self, RSValueMock::Null)
+    }
+
+    fn get_ref(&self) -> Option<&Self> {
+        if let RSValueMock::Reference(boxed) = self {
+            Some(boxed)
+        } else {
+            None
+        }
+    }
+
+    fn get_ref_mut(&mut self) -> Option<&mut Self> {
+        if let RSValueMock::Reference(boxed) = self {
+            Some(boxed.as_mut())
+        } else {
+            None
+        }
+    }
+
+    fn as_str(&self) -> Option<&str> {
+        if let RSValueMock::String(s) = self {
+            Some(s)
+        } else {
+            None
+        }
+    }
+
+    fn as_num(&self) -> Option<f64> {
+        if let RSValueMock::Number(num) = self {
+            Some(*num)
+        } else {
+            None
+        }
+    }
+
+    fn get_type(&self) -> ffi::RSValueType {
+        // Mock implementation, return a dummy type
+        match &self {
+            RSValueMock::Null => ffi::RSValueType_RSValue_Null,
+            RSValueMock::Number(_) => ffi::RSValueType_RSValue_Number,
+            RSValueMock::String(_) => ffi::RSValueType_RSValue_String,
+            RSValueMock::Reference(reference) => reference.get_type(),
+        }
+    }
+
+    fn is_ptr_type() -> bool {
+        // Mock implementation, return false
+        false
+    }
+}
+
+#[test]
+fn test_rssortingvector_creation() {
```

Comment:
```suggestion
fn test_creation() {
```

We're in the `sorting_vector` crate, no need to prefix every test name with it.

Reply:
https://github.com/RediSearch/RediSearch/pull/6311/commits/ff6ffe92abf6b743b628817614a426f5ff6f2ba9